roles/kubernetes-apps/scheduler_plugins/plugins/tasks/main.yml

```diff
@@ -0,0 +1,62 @@
+---
+- name: Scheduler Plugins {{ scheduler_plugin }} | Create dir
+  file:
+    path: "{{ kube_config_dir }}/scheduler-plugins/{{ scheduler_plugin }}"
+    state: directory
+    owner: root
+    group: root
+    mode: 0755
+  when: inventory_hostname == groups['kube_control_plane'][0]
+  tags:
+    - scheduler_plugins
+
+- name: Schedler Plugins | Set scheduler plugin template lists
+  when: inventory_hostname == groups['kube_control_plane'][0]
+  tags:
+    - scheduler_plugins
+  set_fact:
+    scheduler_plugin_templates:
+      CoScheduling:
+        - { name: coscheduling-crd, file: crd.yml }
+      CapacityScheduling:
+        - { name: capacityscheduling-crd, file: crd.yml }
+      NodeResources: []
+
+- name: Scheduler Plugins {{ scheduler_plugin }} | Create manifests
+  template:
+    src: "{{ scheduler_plugin | lower }}/{{ item.file }}.j2"
+    dest: "{{ kube_config_dir }}/scheduler-plugins/{{ scheduler_plugin }}/{{ item.file }}"
+    mode: 0644
+  with_items: "{{ scheduler_plugin_templates[scheduler_plugin] }}"
+  register: coscheduling_manifests
+  when: inventory_hostname == groups['kube_control_plane'][0]
+  tags:
+    - scheduler_plugins
+
+- name: Scheduler Plugins {{ scheduler_plugin }} | Apply manifests
+  kube:
+    name: "{{ item.item.name }}"
+    kubectl: "{{ bin_dir }}/kubectl"
+    filename: "{{ kube_config_dir }}/scheduler-plugins/{{ scheduler_plugin }}/{{ item.item.file }}"
+    state: "latest"
+  with_items: "{{ coscheduling_manifests.results }}"
+  when: inventory_hostname == groups['kube_control_plane'][0]
+  tags:
+    - scheduler_plugins
+
+- name: Scheduler Plugins {{ scheduler_plugin }} | Create scheduler configuration
```

Comment:
can you leverage the existing capability to pass-in the config files ?

https://github.com/kubernetes-sigs/kubespray/pull/7351/files

Reply:
It makes more sense to set a default configuration, but provide the ability to override it, i.e. pass it through group_vars and render it into predefined templates.